src/groovy/org/pih/warehouse/core/ActivityCode.groovy

```diff
@@ -91,6 +94,8 @@ enum ActivityCode {
                 SUBMIT_REQUEST,
                 DYNAMIC_CREATION
         ]
+
+        return CH.config.openboxes.client.autosave.enabled ? activityCodes + [AUTOSAVE] : activityCodes
```

Comment:
I don't like this. A user should be able to set AUTOSAVE on any location at any time. If we disable autosave globally, that overrides the location setting.

Reply:
I think you're right, but won't it be confusing to users, that they set the `AUTOSAVE`, but it does not work for them?. 
@alannadolny will you make/prepare this small change (allow always set `AUTOSAVE`, but needs the global config to be working)?

Reply:
I prepared this small change [here](https://github.com/openboxes/openboxes/pull/3982/files)